src/main/java/ee/tuleva/onboarding/common/FtpClient.java

```diff
@@ -0,0 +1,56 @@
+package ee.tuleva.onboarding.common;
+
+import lombok.*;
+import org.apache.commons.net.*;
+import org.apache.commons.net.ftp.*;
+
+import java.io.*;
+import java.util.*;
+import java.util.stream.*;
+
+@RequiredArgsConstructor
+public class FtpClient {
+    private final String server;
+    private final String user;
+    private final String password;
+    private final int port;
+
+    private FTPClient ftp;
+
+    public void open() throws IOException {
+        ftp = new FTPClient();
+
+        ftp.addProtocolCommandListener(new PrintCommandListener(new PrintWriter(System.out)));
+        ftp.connect(server, port);
+        int reply = ftp.getReplyCode();
+        if (!FTPReply.isPositiveCompletion(reply)) {
+            ftp.disconnect();
+            throw new IOException("Exception in connecting to FTP Server");
+        }
+
+        ftp.login(user, password);
+    }
+
+    public void close() throws IOException {
+        ftp.disconnect();
+    }
+
+    public Collection<String> listFiles(String path) throws IOException {
```

Comment:
This should return a more specific interface. A `List<String>` is better than `Collection<String>`.